### PR TITLE
added commandline args in projectCreate

### PR DESCRIPTION
**Pull request by planetplut0**
please review this

### CODE REVIEW

File: .local/bin/scripts/projectCreate [hunk 1]
```diff
@@ -1,100 +1,189 @@
...
+            git_init='true'
+            ;;
+        l)
+            lang="${OPTARG}"
+            ;;
+        p)
+            project_name="${OPTARG}"
+            ;;
+        c)
+            class_name="${OPTARG}"
```
Comment (RishabhRD) [inline .local/bin/scripts/projectCreate:19]:
```suggestion
            class_name="${OPTARG}"
```
the options like class_name and executable_name varies according to language. Maybe lua/rust or some other language would have some other terminology. So, including that in command line options would be an overhead in terms of long goal. Those can be taken as input interactively from user based on project. That would be better as these are not generalised options.

File: .local/bin/scripts/projectCreate [hunk 2]
```diff
@@ -1,100 +1,189 @@
...
+        h)
+            help
+            ;;
+    esac
+done
+
+echo $git_init
+echo $lang
+echo "$project_name"
+echo "$class_name"
```
Comment (RishabhRD) [inline .local/bin/scripts/projectCreate:33]:
```suggestion
echo $git_init
echo $lang
echo "$project_name"
echo "$class_name"
```
These echoes are redundant. As user provides all this option himself.

File: .local/bin/scripts/projectCreate [hunk 3]
```diff
@@ -1,100 +1,189 @@
...
+            lang="${OPTARG}"
+            ;;
+        p)
+            project_name="${OPTARG}"
+            ;;
+        c)
+            class_name="${OPTARG}"
+            ;;
+        e)
+            exe_name="${OPTARG}"
```
Comment (RishabhRD) [inline .local/bin/scripts/projectCreate:28]: options like class_name and exe_name should not be provided as command-line argument as they are not generalised options and specific to some certain language. Instead we can take these input interactively for different languages (if it is needed). For example, lua/rust may have some other options required to build project then adding those to command line options would increase an overhead for users to learn those options and would be a little bit inconvenient.

File: .local/bin/scripts/projectCreate (hunk 2 above)
Comment (RishabhRD) [inline .local/bin/scripts/projectCreate:33]: These echo statements seem redundant as user provides all these option himself.

File: .local/bin/scripts/projectCreate [hunk 4]
```diff
@@ -1,100 +1,189 @@
...
+  touch $project_name/$_project_name/__main__.py
+  [ "$git_init" == "true" ] && git init $project_name
+
+}
+
+function help(){
+  echo "usage: projectCreate -l lang -p projectName -c className"
+  echo "optional : -g intialise a git repo inside project folder"
+  echo "         : -h show this help message"
+}
```
Comment (RishabhRD) [inline .local/bin/scripts/projectCreate]: i was not able to retrieve this help function from command line. calling -h leads to some random help things. Can you check it out. Or I am missing something in here.

**Comment by RishabhRD** [comment]
Maybe this feature would be an easy one to add: if user doesn't provide project_name or language_name, then we can ask him interactively instead of falling back to help and script would be more convenient to user. Achieving this would be easy I guess as we just need to check project_name and other related variable to be empty or not and ask for it.

Other changes are great. It's lot cleaner in compare to what I developed initially. It's really a good work. With some changes, ready to merge to upstream. :rocket:

**Review by RishabhRD — CHANGES_REQUESTED** [review]
Requested changes:
1-> help option review
2-> replaces redundant echoes with interactive input of generalise option if not present already(project name and language name)
3-> class_name, executable to be interactive for language that needs it as it would be overhead when having more languages.

File: .local/bin/scripts/projectCreate (hunk 2 above)
Comment (RishabhRD) [inline .local/bin/scripts/projectCreate]:
```suggestion
if [[ -z "$lang ]]; then
   .... take input interactively
fi
```
these echoes are redundant as user provides it already. And it doesn't convey much information to user. Can be replaced with interactive user input for generalised options if not present. Context should be more clear in some other comment.

Comment (planetplut0) [reply .local/bin/scripts/projectCreate]: Ahh , these were just for testing purpose and I forgot to remove them , sorry for the trouble 🙃🙃

File: .local/bin/scripts/projectCreate (hunk 3 above)
Comment (planetplut0) [reply .local/bin/scripts/projectCreate:28]: So you are suggesting to just take -g and -l options ?

File: .local/bin/scripts/projectCreate (hunk 4 above)
Comment (planetplut0) [reply .local/bin/scripts/projectCreate]: I fixed that , I should have tested the script completely before opening this pr 😅

File: .local/bin/scripts/projectCreate (hunk 3 above)
Comment (planetplut0) [reply .local/bin/scripts/projectCreate:28]: something like this 
  [[  -z $project_name ]] && printf "Enter project Name : " && read project_name
  [[  -z $class_name ]] && printf "Enter class Name : " && read class_name
(one small request , can you add hacktoberfest topic to your repo  so these prs will count in my hacktoberfest contribution)

Comment (RishabhRD) [reply .local/bin/scripts/projectCreate:28]: yeah exactly like that. -g, -l and -p are options that I think we should support. And for -l and -p we can also have interactive fallback as you mentioned in comment.

File: .local/bin/scripts/projectCreate (hunk 4 above)
Comment (RishabhRD) [reply .local/bin/scripts/projectCreate]: No problem. This is fine. :smile:

File: .local/bin/scripts/projectCreate (hunk 2 above)
Comment (RishabhRD) [reply .local/bin/scripts/projectCreate]: Sure

**Comment by RishabhRD** [comment]
Remaining changes required are:
1-> interactive fallback.
2-> class name and project name interactively only for the language that needs them instead of generalised command line options.

Other things looks great. I don't know about hacktoberfest. But because these are genuine contributions. Not something like a spam. So, I would be glad to add that topic for you. Can you mail me how can I add that topic (hacktoberfest)? :smiley:

File: .local/bin/scripts/projectCreate (hunk 3 above)
Comment (planetplut0) [reply .local/bin/scripts/projectCreate:28]: check this out -> https://github.com/planetplut0/archrice/blob/master/.local/bin/scripts/projectCreate

**Comment by RishabhRD** [comment]
The changes are great! No problem in code :smile: Ready to merge. Great job.

**Pull request merged**